### PR TITLE
Update test_profiling CI runner to Ubuntu 22.04 to re-enable DHAT copy mode test

### PR DESCRIPTION
**Pull request by Kobzol**
Let's see if this works.

### CODE REVIEW

**Comment by Kobzol** [comment]
Hmm, it seems that with newer Ubuntu, Valgrind fails for more profilers than before :D Maybe we should just build a recent version of Valgrind in CI from scratch? @nnethercote @Mark-Simulacrum

**Comment by nnethercote** [comment]
> Valgrind: debuginfo reader: Possibly corrupted debuginfo file.

Right, the released versions of Valgrind can't handle the debuginfo. A from-trunk version should work. Though I don't know why we aren't already getting this error.

**Comment by Kobzol** [comment]
Maybe the old `valgrind` versions didn't have this problem :thinking:

**Comment by Mark-Simulacrum** [comment]
No major objections to building from scratch, it's probably not too slow or too hard to do that here.

**Comment by Kobzol** [comment]
`Valgrind` is now built in CI, it takes about 2 minutes, which seems fine, the `Test profiling` workflow is not on the critical path.

File: ci/check-profiling.sh [hunk 1]
```diff
@@ -95,16 +95,16 @@ grep -q "dhatFileVersion" results/dhout-Test-helloworld-Check-Full
...
-#    profile_local dhat-copy $bindir/rustc \
-#        --id Test \
-#        --profiles Check \
-#        --cargo $bindir/cargo \
-#        --include helloworld \
-#        --scenarios Full
-#test -f results/dhcopy-Test-helloworld-Check-Full
-#grep -q "dhatFileVersion" results/dhcopy-Test-helloworld-Check-Full
+RUST_BACKTRACE=1 RUST_LOG=raw_cargo_messages=trace,collector=debug,rust_sysroot=debug \
+    cargo run -p collector --bin collector -- \
```
Comment (nnethercote) [inline ci/check-profiling.sh:99]: Nice!

**Review by nnethercote — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -131,11 +131,20 @@ jobs:
           toolchain: beta
           override: true
 
-      - name: Install profilers
+      - name: Install Valgrind
```
Comment (nnethercote) [inline .github/workflows/ci.yml:134]: A brief comment about why we build Valgrind ourselves would be useful here.

Comment (Kobzol) [reply .github/workflows/ci.yml:136]: Good idea, I added a comment.

**Pull request merged**